pkg/cmd/analyzer/analyzer.go

```diff
@@ -0,0 +1,150 @@
+	fs.StringVar(&amqpUri, "amqp-uri", "", "Explicit AMQP URI in case of non-default protocols/ports (optional). Must point to the same cluster as rabbitmqUri")
+
+	fs.StringVar(&golivepeerExchange, "golivepeer-exchange", "lp_golivepeer_metadata", "Name of RabbitMQ exchange to bind the stream to on creation")
+	fs.StringVar(&shardPrefixesFlag, "shard-prefixes", "", "Comma-separated list of prefixes of manifest IDs to process events from")
+
+	// Server options
+	fs.StringVar(&serverOpts.Host, "host", "localhost", "Hostname to bind to")
+	fs.UintVar(&serverOpts.Port, "port", 8080, "Port to listen on")
+	fs.DurationVar(&serverOpts.ShutdownGracePeriod, "shutdown-grace-perod", 15*time.Second, "Grace period to wait for server shutdown before using the force")
+	// API Handler
+	fs.StringVar(&serverOpts.APIRoot, "api-root", "/data", "Root path where to bind the API to")
+	fs.StringVar(&serverOpts.AuthURL, "auth-url", "", "Endpoint for an auth server to call for both authentication and authorization of API calls")
+	fs.StringVar(&serverOpts.OwnRegion, "own-region", "", "Identifier of the region where the service is running, used for triggering global request proxying")
+	fs.StringVar(&serverOpts.RegionalHostFormat, "regional-host-format", "localhost", "Format to build regional URL for proxying to other regions. Should contain 1 %s directive where the region will be replaced (e.g. %s.livepeer.monster)")
+
+	// Streaming options
+	fs.StringVar(&streamingOpts.Stream, "rabbitmq-stream-name", "lp_stream_health_v0", "Name of RabbitMQ stream to create and consume from")
+	fs.StringVar(&streamingOpts.ConsumerName, "consumer-name", "", `Consumer name to use when consuming stream (default "analyzer-${hostname}")`)
+	fs.StringVar(&streamingOpts.MaxLengthBytes, "stream-max-length", "50gb", "When creating a new stream, config for max total storage size")
+	fs.StringVar(&streamingOpts.MaxSegmentSizeBytes, "stream-max-segment-size", "500mb", "When creating a new stream, config for max stream segment size in storage")
+	fs.DurationVar(&streamingOpts.MaxAge, "stream-max-age", 30*24*time.Hour, `When creating a new stream, config for max age of stored events`)
+	fs.DurationVar(&memoryRecordsTtl, "memory-records-ttl", 24*time.Hour, `How long to keep data records in memory about inactive streams`)
+
+	flag.Set("logtostderr", "true")
+	glogVFlag := flag.Lookup("v")
+	verbosity := fs.Int("v", 0, "Log verbosity {0-10}")
+
+	fs.String("config", "", "config file (optional)")
+	ff.Parse(fs, os.Args[1:],
+		ff.WithConfigFileFlag("config"),
+		ff.WithConfigFileParser(ff.PlainParser),
+		ff.WithEnvVarPrefix("LP"),
+		ff.WithEnvVarIgnoreCommas(true),
+	)
+	flag.CommandLine.Parse(nil)
+	glogVFlag.Value.Set(strconv.Itoa(*verbosity))
+
+	if streamingOpts.ConsumerName == "" {
+		streamingOpts.ConsumerName = "analyzer-" + hostname()
+	}
+	if shardPrefixesFlag != "" {
+		shardPrefixes = strings.Split(shardPrefixesFlag, ",")
+	}
+}
+
+func Main() {
```

Comment:
WDYT of calling this `Run` and receiving all the static `var`s declared above and initialized with flags on `init`? Then `init` can be kept on the `./cmd/analyzer` package instead and this package will be usable without forcing all the flags into your own cmdline.

Or is it supposed to be re-used like that, with the same CLI flags? Could you give more details on how this is going to be used without the whole CLI?

Reply:
The architecture I'm thinking for Livepeer-in-a-Box v2 is "Busybox for Livepeer." So actual usage of this would look something like:

```
ln -s ./livepeer-in-a-box ./analyzer
./analyzer -help
```

and then it would be identical.

Reply:
Behind the scenes, there'd be a little bit of code that checks `argv[0]` to see how it was invoked and then runs the appropriate `Main()` process.

Reply:
I see. The problem with the current approach then is that I kept all the flag parsing in the `init()` function, which will be automatically executed anytime this package is imported. So we'd be parsing the analyzer CLI even when we were supposed to run another command.

The easiest solution is to just rename `init()` to sth else like `parseFlags()` and call it first thing on the `Main` function. A less easy but maaayybe more elegant solution would be to have `ParseFlags()` be public as well, and return the result of the parsing which is than passed into the `Run(opts)` function. Downside is that caller needs to call 2 functions instead of 1, but I like not exposing a "Main" function which feels weird (although we can also just rename this to `Run` as is). 

I'm fine with either solution, all nitpicks, but the `init()` needs to go away if i got things right.

Reply:
so would all those vars at the top need to move into a singleton struct or something to avoid unnecessarily allocating them when we're not actually invoking the analyzer?

if so... maybe a ParseFlags() config that produces that configuration struct from ff, a Run(config c) that receives that struct, and a Main() that runs as Run(ParseFlags())?

Reply:
Yeah, I think a `ParseFlags() Config/Options/Params` is a good way to go there. 

> and a Main() that runs as Run(ParseFlags())?
Maybe `ParseAndRun()` just to be more explicit and don't call it "main". We could also just have `parseFlags() config` be called from `Run()` itself in case we're not ever using `Run` without calling `ParseFlags` anyway (which I guess is likely). The struct in this case would only work for not always allocating the vars as you mentioned, but also to not communicate between internal functions via static variables which I think would be weird :P